### ISSUE TITLE
Issue: Dependency cruiser dropping dependencies when both import and type-imports exists to the same module.

### ISSUE THREAD

**Issue by fforres** [sverweij/dependency-cruiser#588]
I have a file that 2 imports to the same file, with one of them being a `type-only` import.

e.g

#### `src/domains/Home/analytics.ts`
```TSX
import { CardDisplay } from "@/features/Card/elements";
import type { CardDisplayType } from "@/features/Card/elements";
```

However, after running `depcruise --validate -T json` I see only one entry inside the resulting json file regarding "@/features/Card/elements" and it doesn't have any type information.

#### `cruise_result.json`
```JSON
{
	.....
    {
      "dynamic": false,
      "module": "@/features/Card/elements",
      "moduleSystem": "es6",
      "exoticallyRequired": false,
      "preCompilationOnly": false,
      "resolved": "src/features/Card/elements.tsx",
      "coreModule": false,
      "followable": true,
      "couldNotResolve": false,
      "dependencyTypes": [
        "aliased"
      ],
      "matchesDoNotFollow": false,
      "circular": false,
      "valid": false,
      "rules": [
        {
          "severity": "error",
          "name": "domain-analytics-files-should-not-import-runtime-code"
        }
      ]
    }
    .....
}
```

There's no information on `dependencyTypes` or a `preCompilationOnly` entry on that file, and only one entry in the `rules` array.

My use case is to limit that a file can only have `type-only` imports, and at the same time, I want to alert when an import to non-type-files exists.

My rules for this scenario are the following

```JSON
[
  {
    "name": "domain-analytics-files-should-only-import-types",
    "severity": "error",
    "from": {
      "path": ["src/domains/Home/analytics.ts"]
    },
    "to": {
      "path": ["src/.*"],
      "pathNot": ["src/__generated__/globalTypes.ts"],
      "preCompilationOnly": true
    }
  },
  {
    "name": "domain-analytics-files-should-not-import-runtime-modules",
    "severity": "error",
    "from": {
      "path": ["src/domains/Home/analytics.ts"]
    },
    "to": {
      "path": ["src/.*"],
      "preCompilationOnly": false
    }
  }
]
```

### Expected Behavior

The issue is that having 2 imports (but one being type-only) It seems dependency-cruiser only considers the first one.
Not sure what the behavior here should actually be, mainly as I don't know how rules will behave once I have a duplicated 2 entries on the `dependencies` array for a module, pointing to the same module.

### Current Behavior

Given the file
#### `src/domains/Home/analytics.ts`
```TSX
import { CardDisplay } from "@/features/Card/elements";
import type { CardDisplayType } from "@/features/Card/elements";
```

The output only has a single entry in the `dependencies` array for that module, (I'd expect it should be 2? maybe? with one of them having a dependencyTypes: [`type-only`]? )

### Steps to Reproduce (for bugs)

Not sure if this is a bug, or working as expected. Happy to provide a repro repository if we need one

### Your Environment

- Version used: 11.3.1
- Node version: v16.14.2
- Operating System and version: macOS Monterrey, v12.3.1
- Link to your project: (It is a private repository, but happy to provide a repro if needed)

**Comment by sverweij** [comment]
Hi @fforres wasn't aware this was a pattern that could be used. Dependency-cruiser de-duplicates imports in a module. The key for that currently consists of the module name and the module system. Looks like that needs to be expanded. I'll put it on the road map.

**Comment by fforres** [comment]
I'd be happy to provide any feedback or help here.

I was thinking something to try another case for mixed imports:

```
import { RuntimeThing, type TypeThing } from "./some/module"
```

But it was a bit late to create another test case.

**Comment by sverweij** [comment]
Hi @fforres got a fix in the oven for the original issue (two separate imports of the same module, one of which a module) - see the linked PR. It'll likely be released (or have been released when you're reading this) with version 11.7.0.

The second example (a regular and type import in one import) will go in a separate issue. 

> I did notice that when using _swc_ for parser even with this fix it only detects the regular import - likely because swc
> doesn't focus on types/ type checking yet (I'll check later on).

**Comment by fforres** [comment]
Saw the fix  🙏 

I'm closing this for cleanup purposes ❤️
